src/Build/Logging/BinaryLogger/Postprocessing/EmbeddedContentEventArgs.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Build.Logging
+{
+    internal sealed class EmbeddedContentEventArgs : EventArgs
+    {
+        public EmbeddedContentEventArgs(EmbeddedContentKind contentKind, Stream contentStream, int length)
+        {
+            ContentKind = contentKind;
+            ContentStream = contentStream;
+            Length = length;
+        }
+
+        public EmbeddedContentEventArgs(EmbeddedContentKind contentKind, Stream contentStream)
+        {
+            ContentKind = contentKind;
+            ContentStream = contentStream;
+        }
+
+        public EmbeddedContentKind ContentKind { get; }
+        public Stream ContentStream { get; }
+        public int? Length { get; }
```

Comment:
Shouldn't this be non-nullable so the event handler knows how many bytes it should read? Currently you don't seem to be initializing this field (all instances are new'ed up using the two-parameter constructor).

Reply:
Yes!
This is removed in the extending PR https://github.com/JanKrivanek/msbuild/pull/1/files#diff-73ad6161ceba968f5877a2765742d8bd157286950e90fc66c75daafbc423c4efL26